unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/RPCSocialApiBridge.cs

```diff
@@ -210,12 +278,128 @@ public async UniTask<FriendRequest> RequestFriendshipAsync(string friendUserId,
+        {
+            switch (friendshipEvent.BodyCase)
+            {
+                case FriendshipEventResponse.BodyOneofCase.Request:
+                    return onRequest(friendshipEvent.Request);
+                case FriendshipEventResponse.BodyOneofCase.Accept:
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    var profile =
+                        await userProfileWebInterfaceBridge.RequestFullUserProfileAsync(friendshipEvent.Accept.User.Address,
+                            cancellationToken);
+
+                    OnFriendRequestAccepted?.Invoke(friendshipEvent.Accept.User.Address, profile);
+                    break;
+                case FriendshipEventResponse.BodyOneofCase.Reject:
+                    OnFriendRequestRejected?.Invoke(friendshipEvent.Reject.User.Address);
+                    break;
+                case FriendshipEventResponse.BodyOneofCase.Delete:
+                    OnDeletedByFriend?.Invoke(friendshipEvent.Delete.User.Address);
+                    break;
+                case FriendshipEventResponse.BodyOneofCase.Cancel:
+                    OnFriendRequestCanceled?.Invoke(friendshipEvent.Cancel.User.Address);
+                    break;
+                default:
+                    Debug.LogErrorFormat("Invalid friendship event {0}", friendshipEvent);
+                    throw new ArgumentOutOfRangeException();
+            }
+
+            return null;
+        }
 
-            return new FriendRequest(GetFriendRequestId(friendUserId, createdAt), createdAt, userProfileWebInterfaceBridge.GetOwn().userId, friendUserId, messageBody);
+        private void ProcessAndLogSubscriptionError(SubscribeFriendshipEventsUpdatesResponse error)
+        {
+            switch (error.ResponseCase)
+            {
+                case SubscribeFriendshipEventsUpdatesResponse.ResponseOneofCase.InternalServerError:
+                    Debug.LogErrorFormat("Subscription to friendship events got internal server error {0}", error.InternalServerError.Message);
+                    break;
+                case SubscribeFriendshipEventsUpdatesResponse.ResponseOneofCase.UnauthorizedError:
+                    Debug.LogErrorFormat("Subscription to friendship events got Unauthorized error {0}", error.UnauthorizedError.Message);
+                    break;
+                case SubscribeFriendshipEventsUpdatesResponse.ResponseOneofCase.ForbiddenError:
+                    Debug.LogErrorFormat("Subscription to friendship events got Forbidden error {0}", error.ForbiddenError.Message);
+                    break;
+                case SubscribeFriendshipEventsUpdatesResponse.ResponseOneofCase.TooManyRequestsError:
+                    Debug.LogErrorFormat("Subscription to friendship events got Too many requests error {0}", error.TooManyRequestsError.Message);
+                    break;
+                default: throw new ArgumentOutOfRangeException(nameof(error), error, null);
```

Comment:
So it can also break the loop

Reply:
You can have the `default` without the throw. Anyway we are not considering `warnings as errors` so it wouldnt be the end of the world to have it there.

Reply:
Done